src/app/models/log.ts

```diff
@@ -0,0 +1,174 @@
+                isDisplayedInModal: true,
+                isDisplayedInTable: true,
+                displayTime: true
+            }
+        ),
+        user: new TextModelField(
+            {
+                title: this.language.log_field_user,
+                isDisplayedInModal: true,
+                isDisplayedInTable: true
+            }
+        ),
+        country: new TextModelField(
+            {
+                title: this.language.log_field_country,
+                isDisplayedInModal: true,
+                isDisplayedInTable: true
+            }
+        )
+    };
+
+    constructor() {
+        super();
+    }
+
+    public static apiToModel(logFromApi: any): Log {
+        const newLog = new Log();
+        const url = logFromApi.url;
+        const request = logFromApi.request;
+        const status = logFromApi.http_status;
+
+        // To filter bad logs
+        if (/.+\/households\/get\/.+/.test(url) || /.+\/export/.test(url) || /.+\/location.+/.test(url)
+            || /.+\/distributions\/criteria\/project\/\d+\/number/.test(url) || /.+\/login.+/.test(url)
+            || /.+\/distributions\/beneficiaries\/project\/\d+/.test(url) || /.+\/indicators/.test(url)
+            || /.+\/deactivate-booklets/.test(url) || /.+\/booklets-print/.test(url) || /.+\/vouchers\/scanned/.test(url)) {
+            return null;
+        }
+
+        // Assign all initial fields
+        newLog.set('id', logFromApi.id);
+        newLog.set('date', DateModelField.formatDateTimeFromApi(logFromApi.date));
+        newLog.set('user', logFromApi.mail_user);
+        newLog.set('url', url);
+        newLog.set('request', request);
+        newLog.set('country', JSON.parse(logFromApi.request).__country);
+
+        // Action and Object
+        const controller = /^\w+\\\w+\\(\w+)Controller::(\w+)Action$/g.exec(logFromApi.controller);
+        controller.shift();
+
+        newLog.set('objectOfAction', controller[0]);
+        const action = controller[1].split(/(?=[A-Z])/).join('_').toLowerCase();
+        newLog.set('action', newLog.language['log_' + action]);
+
+        // Tab name
+        if (url.includes('users') || url.includes('donor') || url.includes('organization')) {
+            newLog.set('tabName', 'administrative');
+        } else if ((url.includes('project') || url.includes('distribution')) && !url.includes('households')) {
+            newLog.set('tabName', 'distributions');
+        } else if (url.includes('beneficiaries') || url.includes('households')) {
+            newLog.set('tabName', 'beneficiaries');
+        } else if (url.includes('vouchers') || url.includes('products') || url.includes('booklets') || url.includes('vendors')) {
+            newLog.set('tabName', 'vouchers');
+        } else {
+            newLog.set('tabName', 'other');
+        }
+
+        // Details
+        newLog.set('details', null);
+        let detailString = '';
+        if (action.includes('update') || action.includes('create')) {
```

Comment:
Switch/case to be more readable?

Reply:
i've done a wonderfully awesome regex that catches only those urls